x-pack/plugins/security_solution/public/common/components/events_viewer/selectors/index.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { createSelector } from 'reselect';
+
+import {
+  getTimelineSelector,
+  globalFiltersQuerySelector,
+  globalQuery,
+  globalQuerySelector,
+  timelineQueryByIdSelector,
+} from '../../../store/inputs/selectors';
+import { getTimelineByIdSelector } from '../../../../timelines/store/timeline/selectors';
+
+/**
+ * This selector is invoked with two arguments:
+ * @param state - the state of the store as defined by State in common/store/types.ts
+ * @param id - a timeline id e.g. `detections-page`
+ *
+ * Example:
+ *  `useSelector((state: State) => eventsViewerSelector(state, id))`
+ */
+export const eventsViewerSelector = createSelector(
+  globalFiltersQuerySelector(),
+  getTimelineSelector(),
```

Comment:
This isn't really tied to your PR, but if you don't mind explaining. Was there a reason for putting the createSelector behind a function? `const globalFiltersQuerySelector = () => createSelector(....);` versus just `const globalFiltersQuerySelector = createSelector(....);`?
`

Reply:
Changing all of those now isn't necessary, but thinking we can simplify all of those to just  `= createSelector(...)`

Reply:
If the existing selectors weren't like the example you provided, the arguments to `createSelector` in this PR would look like this:

```ts
export const eventsViewerSelector = createSelector(
  globalFiltersQuerySelector,
  getTimelineSelector,
  globalQuerySelector,
  globalQuery,
  timelineQueryByIdSelector,
  getTimelineByIdSelector,
  (filters, input, query, globalQueries, timelineQuery, timeline) => ({
    // ...
  })
);
```

but the existing selectors all require the function "wrapping" the selector to first be applied (to get to the actual selector), so the arguments to `createSelector` instead look like this:

```ts
export const eventsViewerSelector = createSelector(
  globalFiltersQuerySelector(),
  getTimelineSelector(),
  globalQuerySelector(),
  globalQuery(),
  timelineQueryByIdSelector(),
  getTimelineByIdSelector(),
  (filters, input, query, globalQueries, timelineQuery, timeline) => ({
    //
  })
);
```

This PR, which only creates one new selector, didn't seem like it would benefit from wrapping the selector in a function, so I didn't do that. I briefly considered also refactoring all the existing selectors to remove the wrappers, but it felt like that might work better in a separate PR, after a team discussion.